modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/registry/AppFacRemoteRegistryAccessService.java

```diff
@@ -531,8 +547,121 @@ private AppFactoryResource convertResourceObject( AppFacResource appFacResource)
     	
     }
 
+    /**
+     * Create runtime properties in Kubernetes for applcations
+     *
+     * @param applicationId      id of the application
+     * @param stage              stage of the application
+     * @param key                property key
+     * @param value              property value
+     * @param isSecured          property is secured or not
+     * @param applicationVersion applicationVersion of the application
+     */
+    @Override public void addRuntimeProperty(String applicationKey, String applicationVersion, String stage,
+            String applicationType, String key, String value, boolean isSecured) throws AppFactoryException {
+        String tenantDomain = PrivilegedCarbonContext.getCurrentContext().getTenantDomain();
+
+        if (log.isDebugEnabled()) {
+            log.debug("Adding runtime property for application : " + applicationKey
+                    + " and applicationVersion : " + applicationVersion + " in stage " + stage
+                    + " with property : " + key + " in tenant domain : " + tenantDomain);
+        }
+
+        RuntimeProperty runtimeProperty = new RuntimeProperty();
+        Map<String, String> properties = new HashMap<String, String>();
+
+        boolean isChecked = isSecured;
+
+        //Checked whether the property applicationType is secured
+        if (isChecked) {
+            runtimeProperty.setPropertyType(RuntimeProperty.PropertyType.SECURED);
+            //the value is encoding base64 
```

Comment:
We can see that its encoding. Question would be why? Please answer why in comment.